plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/io/TrinoFileSystem.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.io;
+
+import org.apache.iceberg.io.FileIO;
+
+import java.io.IOException;
+
+public interface TrinoFileSystem
```

Comment:
Consider marking these classes as thread safe or not.  I believe only the provider is thread safe and the rest are considered single threaded

Reply:
Actually, I think that everything should be thread safe, including `TrinoInput` since it only has positioned read methods. The only thing not safe would be the `FileIterator` returned from `FileSystem.listFiles()` but that seems expected.

Did you see something not thread safe?